src/Microsoft.AspNetCore.SystemWebAdapters.SessionState/RemoteSession/RemoteAppSessionStateManager.cs

```diff
@@ -129,8 +125,13 @@ private async Task SetOrReleaseSessionData(ISessionState? state, CancellationTok
         {
             AddSessionCookieToHeader(req, state.SessionID);
 
-            var bytes = _serializer.Serialize(state);
-            req.Content = new ByteArrayContent(bytes);
+            var ms = new MemoryStream();
+
+            await _serializer.SerializeAsync(state, ms, cancellationToken);
+
+            ms.Position = 0;
+
+            req.Content = new StreamContent(ms);
```

Comment:
Perf: At some point we'll want to avoid creating this MemoryStream and buffering everything. Instead we'd create a custom HttpContent and serialized directly to the output stream.

Reply:
Created an issue to track that